libs/basic_execution/include/hpx/basic_execution/register_locks.hpp

```diff
@@ -17,6 +17,7 @@
 
 ///////////////////////////////////////////////////////////////////////////////
 namespace hpx { namespace util {
+
```

Comment:
Last question, what was the rationale for moving `register_locks` into this module? Fewer dependencies?

Reply:
The rationale is that the lock detection is a property of the execution agent, IMHO, as such it makes sense to have that here.